### PR TITLE
Introduces timeslider component

### PR DESCRIPTION
**Pull request by ahennr**
This is a initial check in of the TimeSlider component (see #24). Currently, the following functionality is available:
* Update `ImageWMS` and `TileWMS` layers given in `layers array`, in particular parameter `TIME`
* Calculate the slider's increment, maxValue and minValue depending on selected time unit (currently year and month), the start time and the end time
* Pass a currently selected date to component

image

Open issues:
* Support WFS layers
* Support time ranges

### CODE REVIEW

File: app/controller/panel/TimeSlider.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    });
+                }
+            }
+        });
+    },
+
+    /**
+     * Function to generate tooltip text for slider
+     *
+     * @param {Ext.slider.Thumb} thumb The Thumb that the tip is attached to
```
Comment (marcjansen) [inline app/controller/panel/TimeSlider.js]: typo
```suggestion
     * @param {Ext.slider.Thumb} thumb The thumb that the tip is attached to
```

File: app/controller/panel/TimeSlider.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+     * @return {String} The formatted date as text
+     */
+    getTipText: function (thumb) {
+        var me = this;
```
Comment (marcjansen) [inline app/controller/panel/TimeSlider.js]: No need to alias here, just use `this`, please.

File: app/controller/panel/TimeSlider.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+     *
+     * @param {Number} sliderValue The value of the slider
+     * @return {Date} The date
+     */
+    getDateForSliderValue: function (sliderValue) {
+        var me = this;
+        var view = me.getView();
+        var timeIncrementUnit = view.timeIncrementUnit || 'year';
+        switch (timeIncrementUnit) {
+        case 'month':
```
Comment (marcjansen) [inline app/controller/panel/TimeSlider.js]: I personally do not like the indentation here, but this is a personal thing, I guess.

File: app/view/main/Map.js [hunk 4]
```diff
@@ -48,6 +48,11 @@ Ext.define('CpsiMapview.view.main.Map', {
             listeners: {
                 afterrender: 'initializeMeasureBtn'
             }
+        }, {
+            xtype: 'cmv_timeslider',
+            startDate: new Date(1998, 0, 1),
+            endDate: new Date(Date.now()),
+            layerNames: [] // TODO: fill me
```
Comment (marcjansen) [inline app/view/main/Map.js]: I think the slider can simply guess these by iteration over all layers (util), and seeing if a property is set.

Comment (ahennr) [reply app/view/main/Map.js]: Alright, I'll introduce such an util this in an upcoming PR

**Comment by ahennr** [comment]
Thx for review

File: app/view/panel/TimeSlider.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            flex: 6,
+            width: 250,
+            listeners: {
+                change: 'onTimeChanged',
+                afterrender: 'initTimeSlider'
+            },
+            bind: {
+                value: '{timeRangeObj.value}'
+            },
+            tipText: me.getController().getTipText.bind(me.getController())
```
[the remark this answers is not in the document]
Comment (ahennr) [reply app/view/panel/TimeSlider.js]: You're right, fixed in 1c8bbfcd

File: app/view/panel/TimeSlider.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        var me = this;
+        me.items = [{
+            fieldLabel: 'Time',
+            xtype: 'slider',
+            labelAlign: 'right',
+            flex: 6,
+            width: 250,
+            listeners: {
+                change: 'onTimeChanged',
+                afterrender: 'initTimeSlider'
```
[the remark this answers is not in the document]
Comment (ahennr) [reply app/view/panel/TimeSlider.js]: In 1c8bbfcd, I changed call of `initTimeSlider` to `beforerender`

**Pull request merged**